### PR TITLE
Enabled StyleCop & Fixed All Build Errors/Warnings

### PR DESCRIPTION
**Pull request by ahoefling**
## Summary
- Added necessary files and NuGet packages to enable StyleCop
- Enabled `TreatBuildWarningsAsErrors` for both Debug and Release Modes
- Fixed build warnings with Newtonsoft.json - There needs to be a local web.config file to build correctly, let's make sure we keep that in there even if it isn't used as part of the build.
- Updated the entire module to meet our coding standards set by stylecop

cc: @valadas @bdukes @mitchelsellers

### CODE REVIEW

**Review by mitchelsellers — APPROVED** [review]

**Comment by ahoefling** [comment]
There are all sorts of problems with this pull request 🤦‍♂️🤦‍♂️. Looks like when I pulled the feature/resource-manager branch this morning it didn't pull all the code correctly. I went through and did all the work that @valadas already did. I am in the process of merging them and taking the best of both changesets. Once that is completed I'll do a force push on this PR so the history is all lined up and ready for review

**Comment by ahoefling** [comment]
I really don't know what happened with local repository this morning when I pulled latest, but here we are a PR that only has minor styling changes. After going through the Resource Manager line-by-line I guess I now have a decent understanding of structure if anything else 😁.

Take it or leave it with this PR, the notable changes here are minor styling changes and the addition of a web.config file.

File: DNN Platform/Modules/ResourceManager/Components/Common/Utils.cs [hunk 1]
```diff
@@ -1,7 +1,6 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information
-
 namespace Dnn.Modules.ResourceManager.Components.Common
```
Comment (ahoefling) [inline DNN Platform/Modules/ResourceManager/Components/Common/Utils.cs:4]: Shouldn't this be `DotNetNuke.Modules.ResourceManager.*` for the namespace instead of `Dnn.Modules.ResourceManager.*`. I see this throughout the new module and to the best of my knowledge we agreed we are to use `DotNetNuke.*`

Comment (valadas) [reply DNN Platform/Modules/ResourceManager/Components/Common/Utils.cs:4]: That is a good question... It was there so I am unsure :)

File: DNN Platform/Modules/ResourceManager/Dnn.Modules.ResourceManager.csproj [hunk 2]
```diff
@@ -252,6 +258,18 @@
       <Link>stylecop.json</Link>
```
Comment (valadas) [inline DNN Platform/Modules/ResourceManager/Dnn.Modules.ResourceManager.csproj:258]: I think the stylecop.json is twice in the csproj file

File: DNN Platform/Modules/ResourceManager/Settings.ascx.cs [hunk 3]
```diff
@@ -55,6 +54,7 @@ public override void LoadSettings()
             }
             catch (Exception exc)
             {
+                // Module failed to load
```
Comment (valadas) [inline DNN Platform/Modules/ResourceManager/Settings.ascx.cs:57]: Do we really need this comment ? The line right under it is pretty self explanatory...

File: DNN Platform/Modules/ResourceManager/Settings.ascx.cs [hunk 4]
```diff
@@ -71,6 +71,7 @@ public override void UpdateSettings()
...
```
Comment (valadas) [inline DNN Platform/Modules/ResourceManager/Settings.ascx.cs:74]: Do we really need this comment ? The line right under it is pretty self explanatory...

File: DNN Platform/Modules/ResourceManager/View.ascx.cs [hunk 5]
```diff
@@ -217,6 +215,7 @@ protected void Page_Load(object sender, EventArgs e)
             }
             catch (Exception exc)
             {
+                // Module failed to load
```
Comment (valadas) [inline DNN Platform/Modules/ResourceManager/View.ascx.cs:218]: Do we really need this comment ? The line right under it is pretty self explanatory...

File: DNN Platform/Modules/ResourceManager/web.config [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+		<assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+			<dependentAssembly>
+				<assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30AD4FE6B2A6AEED" culture="neutral" />
+				<bindingRedirect oldVersion="0.0.0.0-10.0.0.0" newVersion="10.0.0.0" />
+			</dependentAssembly>
+			<dependentAssembly>
+				<assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30ad4fe6b2a6aeed" culture="neutral" />
+				<bindingRedirect oldVersion="0.0.0.0-10.0.0.0" newVersion="10.0.0.0" />
+			</dependentAssembly>
+		</assemblyBinding>
```
Comment (valadas) [inline DNN Platform/Modules/ResourceManager/web.config:19]: It looks like the assembly binding is in here twice

**Review by valadas — CHANGES_REQUESTED** [review]
I left a few comments :)

File: DNN Platform/Modules/ResourceManager/web.config (hunk 6 above)
Comment (ahoefling) [reply DNN Platform/Modules/ResourceManager/web.config:19]: yes, it is in there twice. I'll fix it in the next code push

File: DNN Platform/Modules/ResourceManager/View.ascx.cs (hunk 5 above)
Comment (ahoefling) [reply DNN Platform/Modules/ResourceManager/View.ascx.cs:218]: This was left over from the original code, didn't realize you removed it. I'll fix it in the next code push

File: DNN Platform/Modules/ResourceManager/Settings.ascx.cs (hunk 4 above)
Comment (ahoefling) [reply DNN Platform/Modules/ResourceManager/Settings.ascx.cs:74]: This was left over from the original code, didn't realize you removed it. I'll fix it in the next code push

File: DNN Platform/Modules/ResourceManager/Settings.ascx.cs (hunk 3 above)
Comment (ahoefling) [reply DNN Platform/Modules/ResourceManager/Settings.ascx.cs:57]: This was left over from the original code, didn't realize you removed it. I'll fix it in the next code push

File: DNN Platform/Modules/ResourceManager/Dnn.Modules.ResourceManager.csproj (hunk 2 above)
Comment (ahoefling) [reply DNN Platform/Modules/ResourceManager/Dnn.Modules.ResourceManager.csproj:258]: Yes it is, I'll remove the 2nd entry in the next code push

File: DNN Platform/Modules/ResourceManager/Components/Common/Utils.cs (hunk 1 above)
Comment (ahoefling) [reply DNN Platform/Modules/ResourceManager/Components/Common/Utils.cs:4]: cc: @mitchelsellers @bdukes 

I think it would be appropriate for the folx working on the new module to update the namespace. They will have the most experience on fixing any problems that come up. Especially with web api routing issues where the frontend code is communicating with the `DnnApiController`

File: DNN Platform/Modules/ResourceManager/Dnn.Modules.ResourceManager.csproj [hunk 7]
```diff
@@ -248,9 +254,16 @@
     <Content Include="App_LocalResources\EditFolderMapping.ascx.resx" />
   </ItemGroup>
   <ItemGroup>
-    <AdditionalFiles Include="..\..\..\stylecop.json">
-      <Link>stylecop.json</Link>
-    </AdditionalFiles>
+    <Analyzer Include="..\..\..\packages\StyleCop.Analyzers.1.1.118\analyzers\dotnet\cs\StyleCop.Analyzers.CodeFixes.dll" />
+    <Analyzer Include="..\..\..\packages\StyleCop.Analyzers.1.1.118\analyzers\dotnet\cs\StyleCop.Analyzers.dll" />
```
Comment (bdukes) [inline DNN Platform/Modules/ResourceManager/Dnn.Modules.ResourceManager.csproj:258]: These are already included on lines 269-270
```suggestion
```

File: DNN Platform/Modules/ResourceManager/Services/ItemsController.cs [hunk 8]
```diff
@@ -170,7 +179,7 @@ select new
                         IsDefault =
                         m.MappingName == "Standard" || m.MappingName == "Secure" || m.MappingName == "Database",
                         editUrl = this.UserInfo.IsAdmin ?
-                            moduleContext.EditUrl(
+                            this.GetModuleContext().EditUrl(
```
Comment (bdukes) [inline DNN Platform/Modules/ResourceManager/Services/ItemsController.cs:182]: This is going to call `GetModuleContext` for each item in `mappings`, was this change intentional?

**Review by bdukes — APPROVED** [review]

File: DNN Platform/Modules/ResourceManager/Dnn.Modules.ResourceManager.csproj (hunk 7 above)
Comment (ahoefling) [reply DNN Platform/Modules/ResourceManager/Dnn.Modules.ResourceManager.csproj:258]: Another bad merge 🤦‍♂️. I'll accept the suggested change. Thanks!

File: DNN Platform/Modules/ResourceManager/Services/ItemsController.cs (hunk 8 above)
Comment (ahoefling) [reply DNN Platform/Modules/ResourceManager/Services/ItemsController.cs:182]: Thanks for mentioning this, it is not intentional and I'll fix it in the next update

**Review by bdukes — APPROVED** [review]
Once the build completes, this is good to merge

**Review by valadas — APPROVED** [review]
Looks good to me.
